File: hackertracker/src/main/java/com/shortstack/hackertracker/Model/Company.java
package com.shortstack.hackertracker.Model;

import android.text.TextUtils;

import java.io.Serializable;

/**
 * Created with IntelliJ IDEA.
 * User: Whitney Champion
 * Date: 7/15/13
 * Time: 11:17 PM
 */
public class Company implements Serializable {

    private int id;
    private String title;
    private String description;
    private String link;
    private int partner;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public boolean hasLink() {
        return !TextUtils.isEmpty(link);
    }

    public void setPartner(int partner) {
        this.partner = partner;
    }

    public boolean isPartner() {
        return partner == 1;
    }
}

File: hackertracker/src/main/java/com/shortstack/hackertracker/Model/Company.kt
package com.shortstack.hackertracker.Model

import android.database.Cursor
import android.text.TextUtils
import com.google.gson.Gson
import com.orhanobut.logger.Logger
import org.json.JSONObject

import java.io.Serializable

class Company : Serializable {

    var id: Int = 0
    var title: String? = null
    var description: String? = null
    var link: String? = null
    var partner: Int = 0

    val isPartner: Boolean
        get() = partner == 1

    fun hasLink(): Boolean {
        return !TextUtils.isEmpty(link)
    }

    companion object {
        fun CursorToCompany(gson: Gson, cursor: Cursor): Company {
            val obj = JSONObject()

            val totalColumn = cursor.columnCount

            for (i in 0..totalColumn - 1) {
                try {
                    obj.put(cursor.getColumnName(i), cursor.getString(i))
                } catch (e: Exception) {
                    Logger.e(e, "Failed to convert Cursor into JSONObject.")
                }
            }

            return gson.fromJson(obj.toString(), Company::class.java)
        }
    }
}
